frappe/database/query.py

```diff
@@ -302,12 +305,14 @@ def parse_field(field: str):
 			if isinstance(field, Criterion):
 				_fields.append(field)
 			elif isinstance(field, str):
-				if "," in field:
+				if "," in field and "[" not in field:
 					field = field.casefold() if "`" not in field else field
 					field_list = COMMA_PATTERN.split(field)
 					for field in field_list:
 						if _field := field.strip():
 							_fields.append(parse_field(_field))
+				elif "[" in field and "]" in field:
```

Comment:
Let's use better data structures instead of string here?

Something like this maybe?

`fields = [{"reactions": ["idx", "emoji"]}, "name"]`